### ISSUE TITLE
Bug: Incorrect number of steps in lr scheduler steps for ByteDance

### ISSUE THREAD

**Issue by TheRootOf3** [Adamliu1/SNLP_GCW#14]
There is a bug in the following snippet setting the number of scheduler steps in `unclearn_harm.py`. For ByteDance it sets `num_training_steps` to an incorrect value.

```python
# Prepare.
# num_training_steps = args.max_unlearn_steps
lr_scheduler = get_scheduler(
    name="linear",
    optimizer=optimizer,
    num_warmup_steps=0,
    num_training_steps=args.num_epochs * args.epoch_size)
```

Should be:

```python
# Prepare.
# num_training_steps = args.max_unlearn_steps
lr_scheduler = get_scheduler(
    name="linear",
    optimizer=optimizer,
    num_warmup_steps=0,
    num_training_steps=args.num_epochs * args.epoch_size if args.sequential > 0 else args.max_unlearn_steps
)
```

On a separate note, whether we should even use the scheduler is another question.
